src/goto-symex/symex_atomic_section.cpp

```diff
@@ -16,10 +16,9 @@ void goto_symext::symex_atomic_begin(statet &state)
   if(state.guard.is_false())
     return;
 
-  // we don't allow any nesting of atomic sections
-  if(state.atomic_section_id!=0)
-    throw "nested atomic section detected at "+
-      state.source.pc->source_location.as_string();
+  INVARIANT(
+    state.atomic_section_id == 0,
+    "we don't support nesting of atomic sections");
```

Comment:
This should be an exception, it's perfectly possible to write input programs that exhibit this.